File: SothemaGoalManagement.API/Models/Goal.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System;

namespace SothemaGoalManagement.API.Models
{
    public class Goal
    {
        public int Id { get; set; }
        public int ParentGoalId { get; set; }

        public string Description { get; set; }

        public string Status { get; set; }

        public int Weight { get; set; }

        public GoalType GoalType { get; set; }

        public int GoalTypeId { get; set; }

        public string ProjectName { get; set; }

        public AxisInstance AxisInstance { get; set; }

        public int AxisInstanceId { get; set; }

        public DateTime Created { get; set; }

        public ICollection<GoalEvaluation> GoalEvaluations { get; set; }

        public Goal()
        {
            GoalEvaluations = new Collection<GoalEvaluation>();
        }

    }
}

File: SothemaGoalManagement-SPA/src/app/_models/goal.ts
import { GoalType } from "./goalType";
import { AxisInstance } from "./axisInstance";
import { GoalEvaluation } from "./goalEvaluation";

export interface Goal {
  id: number;
  description: string;
  weight: number;
  created: Date;
  goalType: GoalType;
  projectName: string
  axisInstance: AxisInstance;
  status: string;
  completionRate: number;
  parentGoalId: number;
  comment?: string;
  goalEvaluations: GoalEvaluation[];
  goalGrade: string;
  cascaderFullName?: string;
}
